app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -175,7 +178,16 @@ public void onCreate(Bundle savedInstanceState) {
                 //So we're not in landscape mode anymore, but were before. If we had something selected, we 
                 //need to go to the detail screen instead.
                 if(oldActivity != null) {
-                    if(oldActivity.selectedIntent != null) {
+                    Intent intent = this.getIntent();
+
+                    TreeReference selectedRef = SerializationUtil.deserializeFromIntent(intent,
+                            EntityDetailActivity.CONTEXT_REFERENCE, TreeReference.class);
+                    if (selectedRef != null) {
+                        Intent detailIntent = getDetailIntent(selectedRef, null);
+                        startOther = true;
+                        startActivityForResult(detailIntent, CONFIRM_SELECT);
+                    } else if (oldActivity.selectedIntent != null) {
+                        // XXX: old code that didn't work -- PLM

```

Comment:
Not sure why the following code wasn't working. I couldn't inspect oldActivity.selectedIntent in the debugger, so maybe it was being cleaned up somehow...